src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -21,53 +22,93 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     search,
     options,
     optionsDisplayProperty,
+    hasOther,
     default: defaultOption,
   } = parameters
 
-  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [selectedIndex, setSelectedIndex] = useState<any>()
+  const [localOptions, setLocalOptions] = useState<any[]>([])
+  //saving selected value in local state, so that when options are re rendered, we retain this value
+  const [addedOptions, setAddedOptions] = useState<any[]>([])
+
   const { isEditable } = element
 
+  useEffect (() => {
+    const optionsMapped = options.map((option: any, index: number) => {
+      return {
+        key: `${index}_${option}`,
+        text: optionsDisplayProperty ? option[optionsDisplayProperty] : option,
+        value: index,
+      }
+    })
+    //console.log('added options are: ', addedOptions)
+    if (addedOptions && addedOptions.length > 0) {
+      const newOptionList = [...optionsMapped, ...addedOptions]
+      setLocalOptions(newOptionList)
+    } else {
+      setLocalOptions(optionsMapped)
+    }
+  }, [options, addedOptions])
+
   useEffect(() => {
     // This ensures that, if a default is specified, it gets saved on first load
     if (!currentResponse?.text && defaultOption !== undefined) {
-      const optionIndex = getDefaultIndex(defaultOption, options)
+      const optionValue = getDefaultIndex(defaultOption, options)
```

Comment:
Why change this variable name? (Doesn't look like you've changed any functionality)